javaagent/javaagent.gradle

```
@@ -42,8 +42,14 @@ CopySpec isolateSpec(Collection<Project> projectsWithShadowJar) {
 shadowJar {
   dependsOn ':instrumentation:shadowJar'
   dependsOn ':javaagent-exporters:shadowJar'
-  def projectsWithShadowJar = [project(':instrumentation'), project(':javaagent-exporters')]
+  def projectsWithShadowJar = [project(':javaagent-exporters'), project(':instrumentation')]
   with isolateSpec(projectsWithShadowJar)
+
+  // Exclude dependencies pulled in from instrumentation that exist in javaagent-exporters.
+  // Instrumentation is in the agent class loader, which is the parent for the exporter classloader
+  // and will be used, removing duplicates can prevent issues when downstream projects repackage the
+  // jar.
```

the exporter classloader is only used for loading external exporter jar files

```suggestion
  // Exclude class files pulled in from the javaagent-exporters project that are also pulled in from the
  // instrumentation project.
  // Removing duplicates reduces jar size and can prevent issues when downstream projects repackage
  // the jar.
```